zipline/src/commonMain/kotlin/app/cash/zipline/ZiplineException.kt

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright (C) 2022 Block, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package app.cash.zipline
+
+/**
+ * Thrown by [ZiplineService] function calls when the target function threw an exception.
+ *
+ * This is similar to a wrapping exception like Java's `InvocationTargetException`, but the wrapped
+ * exception type is not generally available because its class might not exist in the catching
+ * process.
+ */
+class ZiplineException(
+  message: String?,
+  cause: Throwable?,
+) : RuntimeException(message, cause) {
+  constructor(message: String?) : this(message, null)
+  constructor(cause: Throwable?) : this(null, cause)
```

Comment:
Why this and not primary + default argument values? JNI would have to call the two-arg one with explicit nulls, but you get a nicer Kotlin API.

Also anything accessed through JNI needs listed in the shrinker rules.

Reply:
Just following the old Java conventions like a chump. Lemme do defaults.